### PR TITLE
package layout header contains breadcrumbs now

### PR DESCRIPTION
**Pull request by sabine**
Ty @Clairevanden for the sketch, here's an initial version of package documentation breadcrumbs. Pretty sure that further improvements to the breadcrumb styles I applied here are possible, but I already like it a lot better than before.

It feels like the boilerplate is contracting and settling into place to leave space for the content to breathe.

ETA: screenshots updated 2023-02-02 after applying styles from Figma

|screen|before|after|
|-|-|-|
|sm|Screenshot 2023-02-02 at 10-30-15 Irmin_mem KV Make Backend Branch · irmin 3 5 1 · OCaml Packages|Screenshot 2023-02-02 at 10-30-24 Irmin_mem KV Make Backend Branch · irmin 3 5 1 · OCaml Packages|
|md|Screenshot 2023-02-02 at 10-30-36 Irmin_mem KV Make Backend Branch · irmin 3 5 1 · OCaml Packages|Screenshot 2023-02-02 at 10-30-42 Irmin_mem KV Make Backend Branch · irmin 3 5 1 · OCaml Packages|
|lg| Screenshot 2023-02-02 at 10-30-53 Irmin_mem KV Make Backend Branch · irmin 3 5 1 · OCaml Packages|Screenshot 2023-02-02 at 10-30-56 Irmin_mem KV Make Backend Branch · irmin 3 5 1 · OCaml Packages|
|xl| Screenshot 2023-02-02 at 10-31-02 Irmin_mem KV Make Backend Branch · irmin 3 5 1 · OCaml Packages|Screenshot 2023-02-02 at 10-31-06 Irmin_mem KV Make Backend Branch · irmin 3 5 1 · OCaml Packages|

Package overview:
|before|after|
|-|-|
|Screenshot 2023-02-02 at 10-34-12 irmin 3 5 1 · OCaml Package|Screenshot 2023-02-02 at 10-34-15 irmin 3 5 1 · OCaml Package|

### CODE REVIEW

File: src/ocamlorg_web/lib/handler.ml [hunk 1]
```diff
@@ -578,42 +578,41 @@ let package_doc t kind req =
...
+                in
+                let first_path_item = List.hd doc.module_path in
+                let first_path_item_title =
+                  match first_path_item with
+                  | `Module s | `ModuleType s | `Parameter (_, s) -> s
+                  | `Class s -> s
+                  | `ClassType s -> s
+                in
+                (* NOTE: if it's a standalone page, there is no library path item.
+                  TODO: update this when the docs pipeline provides
```
Comment (cuihtlauac) [inline src/ocamlorg_web/lib/handler.ml]:
```suggestion
            Ocamlorg_frontend.Package_breadcrumbs.Documentation
              (if doc.module_path != [] then
               let module_path_to_breadcrumb_path_item p =
                 match p with
                 | `Module s -> Ocamlorg_frontend.Package_breadcrumbs.Module s
                 | `ModuleType s -> ModuleType s
                 | `Parameter (i, s) -> Parameter (i, s)
                 | `Class s -> Class s
                 | `ClassType s -> ClassType s
               in
               let first_path_item = List.hd doc.module_path in
               let first_path_item_title =
                 match first_path_item with
                 | `Module s | `ModuleType s | `Parameter (_, s) -> s
                 | `Class s -> s
                 | `ClassType s -> s
               in
               (* NOTE: if it's a standalone page, there is no library path
                  item. TODO: update this when the docs pipeline provides
```

File: src/ocamlorg_web/lib/handler.ml [hunk 2]
```diff
@@ -578,42 +578,41 @@ let package_doc t kind req =
...
+                        (fun (t : Ocamlorg_frontend.Navmap.toc) ->
+                          t.title = first_path_item_title)
+                        toc.children)
+                    maptoc
+                in
+                match library_path_item with
+                | Some item -> Library (item.title, List.map module_path_to_breadcrumb_path_item doc.module_path)
+                | None -> Page (first_path_item_title)
+              else Index
+            )
```
Comment (cuihtlauac) [inline src/ocamlorg_web/lib/handler.ml]:
```suggestion
               let library_path_item =
                 List.find_opt
                   (fun (toc : Ocamlorg_frontend.Navmap.toc) ->
                     List.exists
                       (fun (t : Ocamlorg_frontend.Navmap.toc) ->
                         t.title = first_path_item_title)
                       toc.children)
                   maptoc
               in
               match library_path_item with
               | Some item ->
                   Library
                     ( item.title,
                       List.map module_path_to_breadcrumb_path_item
                         doc.module_path )
               | None -> Page first_path_item_title
              else Index)
```

**Review by cuihtlauac — CHANGES_REQUESTED** [review]

**Comment by cuihtlauac** [comment]
I prefer seeing the breadcrumb below the package + version box (as in M) rather than on the right (as L and X).

I also feel there are two many chevrons, I count 1 horizontal right of version and two right, as brackets surrounding “Documentation”

Maybe, some sort of abbreviation mechanism is needed, those breadcrumbs can be really long

**Comment by sabine** [comment]
Always breaking the breadcrumbs so that the module chain can stand on its own line would look like this:

Screenshot 2023-02-02 at 10-50-27 Irmin_mem Make Metadata · irmin 3 5 1 · OCaml Packages

I propose we keep the breadcrumbs on a single line in this patch, and we can have a vote somewhere to see if there is a bigger sentiment wrt wanting the modules to appear on a separate line. It's a tradeoff, because there would be unused whitespace on the right hand side and more vertical space used by the breadcrumbs on wide screens.

**Comment by Clairevanden** [comment]
I agree!. Cos, as you mentioned a lot of dead white space.

**Pull request merged**